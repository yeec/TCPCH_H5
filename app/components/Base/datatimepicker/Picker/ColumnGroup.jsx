import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Column from './Column';

class ColumnGroup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      visible: props.visible || false,
    };
  }

  onValueChange(v, i) {
    const value = this.getValue().concat();

    value[i] = v;
    this.props.onValueChange(value, i);
  }

  getValue() {
    const { children, selectedValue } = this.props;

    if (selectedValue && selectedValue.length) {
      return selectedValue;
    }
    if (!children) {
      return [];
    }
    return children.map((c) => {
      const cc = c.props.children;
      return cc && cc[0] && cc[0][this.props.valueMember];
    });
  }

  render() {
    const {
      prefixCls, className,
      indicatorStyle, pure,
      children, displayMember, valueMember,
    } = this.props;

    const selectedValue = this.getValue();
    const colElements = children.map((col, i) => {
      return (
        <div key={col.key || i} className={`${prefixCls}-column-group-item`}>
          <Column
            pure={pure}
            indicatorStyle={indicatorStyle}
            prefixCls={prefixCls}
            selectedValue={selectedValue[i]}
            displayMember={displayMember}
            valueMember={valueMember}
            onValueChange={value => this.onValueChange(value, i)}
            {...col.props}
            />
        </div>
      );
    });

    return (
      <div className={classnames(className, `${prefixCls}-column-group`)}>
        {colElements}
      </div>
    );
  }
}

ColumnGroup.propTypes = {
  prefixCls: PropTypes.string,
  onValueChange: PropTypes.func,
};

ColumnGroup.defaultProps = {
  prefixCls: 'ryt-picker',
  onValueChange: () => {},
};

export default ColumnGroup;
