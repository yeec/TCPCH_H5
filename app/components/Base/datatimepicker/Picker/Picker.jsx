import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { arrayTreeFilter, formatToInit, formatBackToObject, isArray, hasChildrenObject } from './utils';
import ColumnGroup from './ColumnGroup';
import Cascader from './Cascader';
import Popup from '../Popup';


// 阻止选择器区域的默认事件
function onContainerClick(e) {
  e.stopPropagation();
}

class Picker extends Component {
  constructor(props) {
    super(props);

    const initValue = props.value || props.defaultValue || [];
    const { dataSource } = props;
    let _data = null;
    let _value = null;

    // 针对单列数据源，转换为[[{}]]
    if (dataSource.length && !isArray(dataSource[0]) && !hasChildrenObject(dataSource[0])) {
      _data = [props.dataSource];
      _value = isArray(initValue) ? initValue : [initValue];
    } else {
      _data = dataSource;
      _value = initValue;
    }

    this.state = {
      visible: props.visible || false,
      value: _value,
      data: _data,
      cascade: dataSource.length && !isArray(dataSource[0]) && hasChildrenObject(dataSource[0]),
    };

    this.tempValue = _value;
  }

  componentWillReceiveProps(nextProps) {
    if ('dataSource' in nextProps && nextProps.dataSource !== this.props.dataSource) {
      const { dataSource } = nextProps;
      let _data = null;

      if (dataSource.length && !isArray(dataSource[0]) && !hasChildrenObject(dataSource[0])) {
        _data = [nextProps.dataSource];
      } else {
        _data = nextProps.dataSource;
      }

      this.setState({
        data: _data,
        cascade: dataSource.length && !isArray(dataSource[0]) && hasChildrenObject(dataSource[0]),
      });
    }

    if ('value' in nextProps && nextProps.value !== this.props.value) {
      let _value = null;
      const { dataSource } = nextProps;

      if (dataSource.length && !isArray(dataSource[0]) && !hasChildrenObject(dataSource[0])) {
        _value = isArray(nextProps.value) ? nextProps.value : [nextProps.value];
      } else {
        _value = nextProps.value;
      }

      this.setState({
        value: _value,
        cascade: dataSource.length && !isArray(dataSource[0]) && hasChildrenObject(dataSource[0]),
      });
      this.tempValue = _value;
    }
  }

  onValueChange(value) {
    this.setState({
      value,
    });
    this.props.onChange(value);
  }

  onCancel() {
    const { onCancel } = this.props;
    this.toggle();
    this.setState({
      value: this.tempValue,
    });
    onCancel && onCancel();
  }

  onOk() {
    const { onOk, valueMember, cols } = this.props;
    const { data, cascade } = this.state;
    const value = this.getInitValue();
    this.setState({
      value,
    });
    this.tempValue = value;
    this.toggle();
    let _value = null;
    _value = formatBackToObject(data, value, cascade, valueMember, cols);
    onOk && onOk(_value);
  }

  onMaskClick() {
    const { onMaskClick } = this.props;
    this.onCancel();
    onMaskClick && onMaskClick();
  }

  getInitValue() {
    const data = this.state.data;
    const { valueMember } = this.props;

    const { value } = this.state;

    if (!value || !value.length) {
      if (this.state.cascade) {
        return formatToInit(data[0], valueMember, this.props.cols);
      }
      return data.map(d => (d[0][valueMember]));
    }

    return value;
  }

  // 切换显示状态
  toggle() {
    if (this.props.disabled) {
      return;
    }
    this.setState({
      visible: !this.state.visible,
    });
  }

  handleClick() {
    this.props.onClick();
    !this.props.disabled && this.toggle();
  }

  close(key) {
    this.setState({
      [`${key}`]: false,
    });
  }

  _displayRender(data) {
    const { displayRender, displayMember, displayAddon } = this.props;

    if (typeof displayRender === 'function') {
      return displayRender(data);
    }
    return data.map((v) => {
      return v[displayMember];
    }).join(displayAddon);
  }

  render() {
    const { prefixCls, disabled, className, cancelText, okText, title, placeholder, valueMember, displayMember, displayAddon } = this.props;
    const { data, value } = this.state;

    let PickerCol = null;

    const classes = classnames({
      [`${prefixCls}-container`]: true,
      [className]: !!className,
    });

    const inputCls = classnames({
      [`${prefixCls}-input`]: true,
      [`${prefixCls}-placeholder`]: !value.join(displayAddon),
      [`${prefixCls}-disabled`]: !!disabled,
    });

    const cols = data.map((d) => {
      return { props: { children: d } };
    });

    if (this.state.cascade) {
      PickerCol = (
        <Cascader
          prefixCls={prefixCls}
          data={data}
          value={this.state.value}
          cols={this.props.cols}
          displayMember={displayMember}
          valueMember={valueMember}
          onChange={v => this.onValueChange(v)}
          />
      );
    } else {
      PickerCol = (
        <ColumnGroup
          className={className}
          prefixCls={prefixCls}
          displayMember={displayMember}
          valueMember={valueMember}
          selectedValue={value}
          onValueChange={v => this.onValueChange(v)} >
          {cols}
        </ColumnGroup>
      );
    }

    const display = () => {
      if (this.state.cascade) {
        if (value.length) {
          const treeChildren = arrayTreeFilter(this.props.dataSource, (item, level) => {
            return item[valueMember] === value[level];
          });

          return this._displayRender(treeChildren);
        }
      }

      const treeChildren2 = data.map((d, index) => {
        if (value[index]) {
          return d.find(obj => value[index] === obj[valueMember]);
        }
        return undefined;
      }).filter(t => !!t);
      return this._displayRender(treeChildren2);
    };

    return (
      <div className={`${prefixCls}`} onClick={() => this.handleClick()}>
        <div className={inputCls}>
          <input type="hidden" value={this.state.value} />
          {display() || placeholder}
        </div>
        <div className={classes} onClick={e => onContainerClick(e)}>
          <Popup
            className="za-popup-inner"
            visible={this.state.visible}
            onMaskClick={() => this.onMaskClick()}>
            <div className={`${prefixCls}-wrapper`}>
              <div className={`${prefixCls}-header`}>
                <div className={`${prefixCls}-cancel`} onClick={() => this.onCancel()}>{cancelText}</div>
                <div className={`${prefixCls}-title`}>{title}</div>
                <div className={`${prefixCls}-submit`} onClick={() => this.onOk()}>{okText}</div>
              </div>
              <div className={`${prefixCls}-mask-top`}>
                <div className={`${prefixCls}-mask-bottom`}>
                  {PickerCol}
                </div>
              </div>
            </div>
          </Popup>
        </div>
      </div>
    );
  }
}

Picker.propTypes = {
  visible: PropTypes.bool,
  placeholder: PropTypes.string,
  title: PropTypes.string,
  cancelText: PropTypes.string,
  okText: PropTypes.string,
  displayAddon: PropTypes.string,
  disabled: PropTypes.bool,
  dataSource: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.object, PropTypes.array])).isRequired,
  cols: PropTypes.number,
  onOk: PropTypes.func.isRequired,
  onCancel: PropTypes.func,
  onMaskClick: PropTypes.func,
  displayRender: PropTypes.func,
  prefixCls: PropTypes.string,
  displayMember: PropTypes.string,
  valueMember: PropTypes.string,
};

Picker.defaultProps = {
  visible: false,
  placeholder: '请选择',
  title: '请选择',
  cancelText: '取消',
  okText: '确定',
  displayAddon: '',
  disabled: false,
  dataSource: [],
  onClick: () => {},
  onChange: () => {},
  onOk: () => {},
  onCancel: () => {},
  onMaskClick: () => {},
  prefixCls: 'ryt-picker',
  displayMember: 'label',
  valueMember: 'value',
};

export default Picker;
