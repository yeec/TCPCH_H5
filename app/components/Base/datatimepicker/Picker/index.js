import Picker from './Picker';
import Date from './DatePicker';
import Stack from './StackPicker';
import './style/index.web';
Picker.Date = Date;
Picker.Stack = Stack;

export default Picker;
