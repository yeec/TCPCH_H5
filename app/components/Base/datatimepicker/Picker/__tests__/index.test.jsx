import React from 'react';
import { render, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Picker from '../index';

describe('Picker', () => {
  it('Picker', () => {
    const wrapper = render(
      <Picker
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Cascader Picker', () => {
    jest.useFakeTimers();
    const wrapper = mount(
      <Picker
        dataSource={[
          {
            value: '1',
            label: '选项一',
            children: [
              { value: '11', label: '选项一' },
              { value: '12', label: '选项二' },
            ],
          },
          {
            value: '2',
            label: '选项一',
            children: [
              { value: '21', label: '选项一' },
              { value: '22', label: '选项二' },
            ],
          },
        ]}
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({ value: ['1', '12'] });
    jest.runAllTimers();
    wrapper.unmount();
  });

  it('DatePicker', () => {
    jest.useFakeTimers();
    const wrapper = mount(
      <Picker.Date mode="date" />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({ value: '2017-09-06' });
    jest.runAllTimers();
    wrapper.unmount();
  });

  it('StackPicker', () => {
    jest.useFakeTimers();
    const wrapper = mount(
      <Picker.Stack
        dataSource={[
          {
            value: '1',
            label: '北京市',
            children: [
              { value: '11', label: '海淀区' },
              { value: '12', label: '西城区' },
            ],
          },
          {
            value: '2',
            label: '上海市',
            children: [
              { value: '21', label: '黄埔区' },
              { value: '22', label: '虹口区' },
            ],
          },
        ]}
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({ value: ['1', '12'] });
    jest.runAllTimers();
    wrapper.unmount();
  });

  it('receive new dataSourcea', () => {
    const wrapper = shallow(
      <Picker
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        />
    );
    wrapper.setProps({
      dataSource: [
        { value: 'a', label: '选项一' },
        { value: 'b', label: '选项二' },
        { value: 'c', label: '选项三' },
      ],
    });
  });

  it('receive new value', () => {
    const wrapper = shallow(
      <Picker
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        />
    );
    wrapper.setProps({ value: '1' });
  });

  // it('onChange', () => {
  //   const onChange = jest.fn();
  //   const wrapper = mount(
  //     <Picker
  //       dataSource={[
  //         { value: '1', label: '选项一' },
  //         { value: '2', label: '选项二' },
  //       ]}
  //       onChange={onChange}
  //       />
  //   );
  //   wrapper.find('input').simulate('change', { target: { value: '2' } });
  //   expect(onChange).toBeCalled();
  // });
});
