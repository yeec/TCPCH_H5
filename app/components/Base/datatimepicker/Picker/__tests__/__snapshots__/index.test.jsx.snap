// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker Cascader Picker 1`] = `
<Picker
  cancelText="取消"
  dataSource={
    Array [
      Object {
        "children": Array [
          Object {
            "label": "选项一",
            "value": "11",
          },
          Object {
            "label": "选项二",
            "value": "12",
          },
        ],
        "label": "选项一",
        "value": "1",
      },
      Object {
        "children": Array [
          Object {
            "label": "选项一",
            "value": "21",
          },
          Object {
            "label": "选项二",
            "value": "22",
          },
        ],
        "label": "选项一",
        "value": "2",
      },
    ]
  }
  disabled={false}
  displayAddon=""
  displayMember="label"
  okText="确定"
  onCancel={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onMaskClick={[Function]}
  onOk={[Function]}
  placeholder="请选择"
  prefixCls="za-picker"
  title="请选择"
  valueMember="value"
  visible={false}
>
  <div
    className="za-picker"
    onClick={[Function]}
  >
    <div
      className="za-picker-input za-picker-placeholder"
    >
      <input
        type="hidden"
        value={Array []}
      />
      请选择
    </div>
    <div
      className="za-picker-container"
      onClick={[Function]}
    >
      <Popup
        animationDuration={200}
        autoClose={false}
        className="za-popup-inner"
        direction="bottom"
        duration={3000}
        mask={true}
        maskType="normal"
        onMaskClick={[Function]}
        prefixCls="za-popup"
        visible={false}
      >
        <div
          className="za-popup za-popup-hidden za-popup-inner"
        >
          <div
            className="za-popup-wrapper za-popup-wrapper-bottom"
            style={
              Object {
                "WebkitTransitionDuration": "200ms",
                "transitionDuration": "200ms",
              }
            }
          >
            <div
              className="za-picker-wrapper"
            >
              <div
                className="za-picker-header"
              >
                <div
                  className="za-picker-cancel"
                  onClick={[Function]}
                >
                  取消
                </div>
                <div
                  className="za-picker-title"
                >
                  请选择
                </div>
                <div
                  className="za-picker-submit"
                  onClick={[Function]}
                >
                  确定
                </div>
              </div>
              <div
                className="za-picker-mask-top"
              >
                <div
                  className="za-picker-mask-bottom"
                >
                  <Cascader
                    data={
                      Array [
                        Object {
                          "children": Array [
                            Object {
                              "label": "选项一",
                              "value": "11",
                            },
                            Object {
                              "label": "选项二",
                              "value": "12",
                            },
                          ],
                          "label": "选项一",
                          "value": "1",
                        },
                        Object {
                          "children": Array [
                            Object {
                              "label": "选项一",
                              "value": "21",
                            },
                            Object {
                              "label": "选项二",
                              "value": "22",
                            },
                          ],
                          "label": "选项一",
                          "value": "2",
                        },
                      ]
                    }
                    disabled={false}
                    displayMember="label"
                    onChange={[Function]}
                    prefixCls="za-picker"
                    value={Array []}
                    valueMember="value"
                  >
                    <ColumnGroup
                      displayMember="label"
                      onValueChange={[Function]}
                      prefixCls="za-picker"
                      selectedValue={
                        Array [
                          "1",
                          "11",
                        ]
                      }
                      valueMember="value"
                    >
                      <div
                        className="za-picker-column-group"
                      >
                        <div
                          className="za-picker-column-group-item"
                        >
                          <Column
                            displayMember="label"
                            onValueChange={[Function]}
                            prefixCls="za-picker"
                            pure={true}
                            selectedValue="1"
                            valueMember="value"
                          >
                            <div
                              className="za-picker-cascader"
                            >
                              <div
                                className="za-picker-cascader-indicator"
                              />
                              <div
                                className="za-picker-cascader-content"
                              >
                                <div
                                  className="za-picker-cascader-item za-picker-cascader-item-selected"
                                >
                                  选项一
                                </div>
                                <div
                                  className="za-picker-cascader-item"
                                >
                                  选项一
                                </div>
                              </div>
                            </div>
                          </Column>
                        </div>
                        <div
                          className="za-picker-column-group-item"
                        >
                          <Column
                            displayMember="label"
                            onValueChange={[Function]}
                            prefixCls="za-picker"
                            pure={true}
                            selectedValue="11"
                            valueMember="value"
                          >
                            <div
                              className="za-picker-cascader"
                            >
                              <div
                                className="za-picker-cascader-indicator"
                              />
                              <div
                                className="za-picker-cascader-content"
                              >
                                <div
                                  className="za-picker-cascader-item za-picker-cascader-item-selected"
                                >
                                  选项一
                                </div>
                                <div
                                  className="za-picker-cascader-item"
                                >
                                  选项二
                                </div>
                              </div>
                            </div>
                          </Column>
                        </div>
                      </div>
                    </ColumnGroup>
                  </Cascader>
                </div>
              </div>
            </div>
          </div>
          <Mask
            className=""
            onClose={[Function]}
            prefixCls="za-mask"
            style={
              Object {
                "WebkitAnimationDuration": "200ms",
                "animationDuration": "200ms",
              }
            }
            type="normal"
            visible={false}
          />
        </div>
      </Popup>
    </div>
  </div>
</Picker>
`;

exports[`Picker DatePicker 1`] = `
<DatePicker
  cancelText="取消"
  defaultValue=""
  disabled={false}
  displayMember="label"
  locale={
    Object {
      "day": "日",
      "hour": "时",
      "minute": "分",
      "month": "月",
      "year": "年",
    }
  }
  minuteStep={1}
  mode="date"
  okText="确定"
  onCancel={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onMaskClick={[Function]}
  onOk={[Function]}
  placeholder="请选择"
  prefixCls="za-picker"
  title="请选择"
  value=""
  valueMember="value"
  visible={false}
>
  <div
    className="za-picker"
    onClick={[Function]}
  >
    <div
      className="za-picker-input za-picker-placeholder"
    >
      <input
        type="hidden"
        value=""
      />
      请选择
    </div>
    <div
      className="za-picker-container za-picker-hidden"
      onClick={[Function]}
    >
      <Popup
        animationDuration={200}
        autoClose={false}
        direction="bottom"
        duration={3000}
        mask={true}
        maskType="normal"
        onMaskClick={[Function]}
        prefixCls="za-popup"
        visible={false}
      >
        <div
          className="za-popup za-popup-hidden"
        >
          <div
            className="za-popup-wrapper za-popup-wrapper-bottom"
            style={
              Object {
                "WebkitTransitionDuration": "200ms",
                "transitionDuration": "200ms",
              }
            }
          >
            <div
              className="za-picker-wrapper"
            >
              <div
                className="za-picker-header"
              >
                <div
                  className="za-picker-cancel"
                  onClick={[Function]}
                >
                  取消
                </div>
                <div
                  className="za-picker-title"
                >
                  请选择
                </div>
                <div
                  className="za-picker-submit"
                  onClick={[Function]}
                >
                  确定
                </div>
              </div>
              <div
                className="za-picker-mask-top"
              >
                <div
                  className="za-picker-mask-bottom"
                >
                  <ColumnGroup
                    disabled={false}
                    displayMember="label"
                    onValueChange={[Function]}
                    prefixCls="za-picker"
                    selectedValue={
                      Array [
                        "2000",
                        "0",
                        "1",
                      ]
                    }
                    valueMember="value"
                  >
                    <div
                      className="za-picker-column-group"
                    >
                      <div
                        className="za-picker-column-group-item"
                      >
                        <Column
                          displayMember="label"
                          onValueChange={[Function]}
                          prefixCls="za-picker"
                          pure={true}
                          selectedValue="2000"
                          valueMember="value"
                        >
                          <div
                            className="za-picker-cascader"
                          >
                            <div
                              className="za-picker-cascader-indicator"
                            />
                            <div
                              className="za-picker-cascader-content"
                            >
                              <div
                                className="za-picker-cascader-item za-picker-cascader-item-selected"
                              >
                                2000年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2001年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2002年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2003年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2004年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2005年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2006年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2007年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2008年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2009年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2010年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2011年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2012年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2013年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2014年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2015年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2016年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2017年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2018年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2019年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2020年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2021年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2022年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2023年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2024年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2025年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2026年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2027年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2028年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2029年
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2030年
                              </div>
                            </div>
                          </div>
                        </Column>
                      </div>
                      <div
                        className="za-picker-column-group-item"
                      >
                        <Column
                          displayMember="label"
                          onValueChange={[Function]}
                          prefixCls="za-picker"
                          pure={true}
                          selectedValue="0"
                          valueMember="value"
                        >
                          <div
                            className="za-picker-cascader"
                          >
                            <div
                              className="za-picker-cascader-indicator"
                            />
                            <div
                              className="za-picker-cascader-content"
                            >
                              <div
                                className="za-picker-cascader-item za-picker-cascader-item-selected"
                              >
                                1月
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2月
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                3月
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                4月
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                5月
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                6月
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                7月
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                8月
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                9月
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                10月
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                11月
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                12月
                              </div>
                            </div>
                          </div>
                        </Column>
                      </div>
                      <div
                        className="za-picker-column-group-item"
                      >
                        <Column
                          displayMember="label"
                          onValueChange={[Function]}
                          prefixCls="za-picker"
                          pure={true}
                          selectedValue="1"
                          valueMember="value"
                        >
                          <div
                            className="za-picker-cascader"
                          >
                            <div
                              className="za-picker-cascader-indicator"
                            />
                            <div
                              className="za-picker-cascader-content"
                            >
                              <div
                                className="za-picker-cascader-item za-picker-cascader-item-selected"
                              >
                                1日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                2日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                3日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                4日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                5日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                6日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                7日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                8日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                9日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                10日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                11日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                12日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                13日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                14日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                15日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                16日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                17日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                18日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                19日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                20日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                21日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                22日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                23日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                24日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                25日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                26日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                27日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                28日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                29日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                30日
                              </div>
                              <div
                                className="za-picker-cascader-item"
                              >
                                31日
                              </div>
                            </div>
                          </div>
                        </Column>
                      </div>
                    </div>
                  </ColumnGroup>
                </div>
              </div>
            </div>
          </div>
          <Mask
            className=""
            onClose={[Function]}
            prefixCls="za-mask"
            style={
              Object {
                "WebkitAnimationDuration": "200ms",
                "animationDuration": "200ms",
              }
            }
            type="normal"
            visible={false}
          />
        </div>
      </Popup>
    </div>
  </div>
</DatePicker>
`;

exports[`Picker Picker 1`] = `
<div
  class="za-picker"
>
  <div
    class="za-picker-input za-picker-placeholder"
  >
    <input
      type="hidden"
      value=""
    />
    请选择
  </div>
  <div
    class="za-picker-container"
  >
    <div
      class="za-popup za-popup-hidden za-popup-inner"
    >
      <div
        class="za-popup-wrapper za-popup-wrapper-bottom"
        style="-webkit-transition-duration:200ms;transition-duration:200ms;"
      >
        <div
          class="za-picker-wrapper"
        >
          <div
            class="za-picker-header"
          >
            <div
              class="za-picker-cancel"
            >
              取消
            </div>
            <div
              class="za-picker-title"
            >
              请选择
            </div>
            <div
              class="za-picker-submit"
            >
              确定
            </div>
          </div>
          <div
            class="za-picker-mask-top"
          >
            <div
              class="za-picker-mask-bottom"
            >
              <div
                class="za-picker-column-group"
              >
                <div
                  class="za-picker-column-group-item"
                >
                  <div
                    class="za-picker-cascader"
                  >
                    <div
                      class="za-picker-cascader-indicator"
                    />
                    <div
                      class="za-picker-cascader-content"
                    >
                      <div
                        class="za-picker-cascader-item za-picker-cascader-item-selected"
                      >
                        选项一
                      </div>
                      <div
                        class="za-picker-cascader-item"
                      >
                        选项二
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Picker StackPicker 1`] = `
<PickerStack
  cols={Infinity}
  dataSource={
    Array [
      Object {
        "children": Array [
          Object {
            "label": "海淀区",
            "value": "11",
          },
          Object {
            "label": "西城区",
            "value": "12",
          },
        ],
        "label": "北京市",
        "value": "1",
      },
      Object {
        "children": Array [
          Object {
            "label": "黄埔区",
            "value": "21",
          },
          Object {
            "label": "虹口区",
            "value": "22",
          },
        ],
        "label": "上海市",
        "value": "2",
      },
    ]
  }
  disabled={false}
  displayItems={8}
  displayRender={[Function]}
  itemHeight={50}
  itemRender={[Function]}
  labelAddon=" > "
  onCancel={[Function]}
  onOk={[Function]}
  placeholder="请选择"
  prefixCls="za-picker"
  title="请选择"
  validate={[Function]}
  value={Array []}
  valueMember="value"
>
  <div
    className="za-picker"
  >
    <div
      className="za-picker-input za-picker-placeholder"
      onClick={[Function]}
    >
      <input
        type="hidden"
        value={Array []}
      />
      请选择
    </div>
    <div
      className="za-picker-container za-picker-hidden"
      onClick={[Function]}
    >
      <Popup
        animationDuration={200}
        autoClose={false}
        direction="bottom"
        duration={3000}
        mask={true}
        maskType="normal"
        onMaskClick={[Function]}
        prefixCls="za-popup"
        visible={false}
      >
        <div
          className="za-popup za-popup-hidden"
        >
          <div
            className="za-popup-wrapper za-popup-wrapper-bottom"
            style={
              Object {
                "WebkitTransitionDuration": "200ms",
                "transitionDuration": "200ms",
              }
            }
          >
            <div
              className="za-picker-wrapper"
            >
              <div
                className="za-picker-header"
              >
                <div
                  className="za-picker-cancel"
                  onClick={[Function]}
                >
                  取消
                </div>
                <div
                  className="za-picker-title"
                >
                  请选择
                </div>
                <div
                  className="za-picker-submit"
                />
              </div>
              <div
                className="za-picker-crumbs"
              >
                <p>
                  选择：
                </p>
              </div>
              <div
                className="za-picker-stack-group"
              >
                <div
                  className="za-picker-stack-column lower-hidden"
                  onClick={[Function]}
                >
                  <div
                    className="za-picker-stack-column-wrapper"
                    onClick={[Function]}
                  >
                    <div
                      className="za-picker-stack-item"
                      onClick={[Function]}
                    >
                      北京市
                    </div>
                    <div
                      className="za-picker-stack-item"
                      onClick={[Function]}
                    >
                      上海市
                    </div>
                  </div>
                </div>
                <div
                  className="za-picker-stack-column lower-hidden"
                  onClick={[Function]}
                >
                  <div
                    className="za-picker-stack-column-wrapper"
                    onClick={[Function]}
                  >
                    <div
                      className="za-picker-stack-item"
                      onClick={[Function]}
                    >
                      海淀区
                    </div>
                    <div
                      className="za-picker-stack-item"
                      onClick={[Function]}
                    >
                      西城区
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <Mask
            className=""
            onClose={[Function]}
            prefixCls="za-mask"
            style={
              Object {
                "WebkitAnimationDuration": "200ms",
                "animationDuration": "200ms",
              }
            }
            type="normal"
            visible={false}
          />
        </div>
      </Popup>
    </div>
  </div>
</PickerStack>
`;
